### PR TITLE
Add new "examples" parameter to Create & CreateProjectMojo 

### LINKED ISSUE #12957: Add a way to select a specific example codestarts but still add extensions
**Description**
This may require a bit of work in the api..

### PR DESCRIPTION
**Pull request by ia3andy**
Closes #12957

This PR also change the defaults when creating a project with the Quarkus CLI (to follow other Quarkus tools):
- use `code-with-quarkus` as default project name
- use `1.0.0-SNAPSHOT` as default project version

### CODE REVIEW

**Comment by gsmet** [comment]
I suspect the CI failures are related.

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java [hunk 1]
```diff
@@ -21,16 +21,20 @@
...
-    String version = "1.0-SNAPSHOT";
+    String version = "1.0.0-SNAPSHOT";
 
     @CommandLine.Option(names = { "-0",
             "--no-examples" }, order = 4, description = "Generate without example code.")
     boolean noExamples = false;
 
+    @CommandLine.Option(names = { "-e",
+            "--examples" }, order = 4, description = "Choose which example(s) you want in the generated Quarkus application.")
+    Set<String> examples;
```
Comment (gastaldi) [inline devtools/cli/src/main/java/io/quarkus/cli/Create.java:36]: Codestarts are now called examples? This is a bit confusing

Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:36]: Well, for a non dev user there is no such notion of codestarts in code.quarkus.io or in any of the tooling. Also the existing option was called `noExamples`.

Also the kind of codestarts we are talking here are codestarts of type `code` and tagged `example`. 

By default, this is dealt thanks to the extension codestart mapping, but this option let you override the default to choose your own set of examples.

Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:36]: cc @maxandersen

Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:36]: @gastaldi 👆

**Comment by ia3andy** [comment]
The CI error is now unrelated.

**Review by gastaldi — APPROVED** [review]
LGTM

**Pull request merged**